utils/submit_track_templates.py

```diff
@@ -0,0 +1,230 @@
+    exit(1)
+  # check for gene track descriptions input
+  if not args.csv:
+    args.csv = f"{template_dir}/beta2-gene-desc.csv"
+  if not os.path.isfile(args.csv):
+    print(f"Error: gene track CSV file not found: {args.csv}")
+    exit(1)
+  parse_csv(args.csv)
+    
+
+def log(msg: object) -> None:
+  if not args.quiet: 
+    print(msg)
+
+
+def parse_csv(path):
+  global gene_desc
+  with open(path) as f:
+    reader = csv.DictReader(f)
+    for line in reader:
+      gene_desc[line['Genome_UUID']] = {
+        'method': line['Annotated_imported'], 'source': line['Source_name'], 'url': line['Source_URL']
+      }
+
+
+# 1) Loop through all the available bigbed/bigwig datafiles
+def process_data_dir() -> None:
+  if not os.path.isdir(data_dir):
+    print(f"Error: data directory {data_dir} not found")
+    exit(1)
+  subdirs = os.listdir(data_dir)
+  i = 0
+  total = len(args.genome or subdirs)
+  for subdir in subdirs:
+    if not os.path.isdir(f"{data_dir}/{subdir}"):
+      continue
+    try:
+      UUID(subdir, version=4)
+    except ValueError:
+      log(f"Skipping non-genome directory {subdir}")
+      continue
+    if args.genome and subdir not in args.genome:
+      continue
+    i += 1
+    if args.resume:
+      if subdir != args.resume:
+        log(f"Skipping genome {subdir} ({i}/{total})")
+        continue
+      else:
+        args.resume = None
+    log(f"Processing genome {subdir} ({i}/{total})")
+    if args.overwrite: # delete existing tracks first
+      delete_tracks(subdir)
+    for file in os.listdir(f"{data_dir}/{subdir}"):
+      if file.endswith(".bb") or file.endswith(".bw"):
+        match_template(subdir, file)
+
+
+# 1b) or use a list of tracks (genomes+template names) instead
+def process_track_list() -> None:
+  files = args.template or args.file
+  total = len(args.genome)
+  for i, genome_id in enumerate(args.genome):
+    if args.resume:
+      if genome_id != args.resume:
+        log(f"Skipping genome {genome_id} ({i+1}/{total})")
+        continue
+      else:
+        args.resume = None
+    log(f"Processing genome {genome_id} ({i+1}/{total})")
+    if args.overwrite: # delete existing tracks first
+      delete_tracks(genome_id)
+    for filename in files:
+      match_template(genome_id, filename)
+
+
+# 2) Load the corresponding track payload template(s)
+def match_template(genome_id: str, datafile: str) -> None:
+  if args.file and not args.template and datafile not in args.file:
+    return
+  filename = os.path.splitext(datafile)[0]
+  if f"{filename}.yaml" in template_files: # 1-to-1 match
+    apply_template(genome_id, filename)
+    return
+  # partial match (multiple tracks per datafile or vice versa)
+  match = False
+  for template_file in template_files:
+    if template_file.startswith(filename):
+      apply_template(genome_id, template_file)
+      match = True
+    elif not match and filename.startswith(os.path.splitext(template_file)[0]):
+      print(f"match to {template_file}")
```

Comment:
I am very confused by what's happening here, especially since the previous line says "if not match", and the print statement says "match". I also don't understand how the if and the elif branch in the loop interact through the `match` variable.

`match to` -> `match for`?

Reply:
The two if statements address (mutually exclusive) use cases/matching strategies (multiple templates match a single datafile or multiple datafiles match a single template).
I renamed some variables, removed debug print and added comments to make it clearer.